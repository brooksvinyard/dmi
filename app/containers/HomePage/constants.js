/*
 * HomePageConstants
 */

export const GET_STRINGS = 'dmi/HomePage/GET_STRINGS';
