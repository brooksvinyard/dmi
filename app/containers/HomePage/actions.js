/*
 * Home Actions
 *
 */

import { GET_STRINGS } from './constants';

/**
 * Gets the strings from the database
 *
 * @return {object} An action object with a type of GET_STRINGS
 */
export function getStrings() {
  return {
    type: GET_STRINGS,
  };
}
