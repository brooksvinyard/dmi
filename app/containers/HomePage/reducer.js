/*
 * HomePage Reducer
 *
 */

import produce from 'immer';
import { GET_STRINGS } from './constants';

// The initial state of the App
export const initialState = {};

/* eslint-disable default-case, no-param-reassign */
const homeReducer = (state = initialState, action) =>
  // eslint-disable-next-line no-unused-vars
  produce(state, draft => {
    switch (action.type) {
      case GET_STRINGS:
        break;
    }
  });

export default homeReducer;
