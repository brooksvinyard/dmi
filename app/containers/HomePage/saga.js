/**
 * Gets the strings from the database
 */

// eslint-disable-next-line no-unused-vars
import { call, put, select, takeLatest } from 'redux-saga/effects';
import { LOAD_STRINGS } from 'containers/App/constants';
import { stringsLoaded, stringsLoadingError } from 'containers/App/actions';

import request from 'utils/request';

/**
 * Localhost Strings request/response handler
 */
export function* getStrings() {
  const requestURL = `http://localhost:3000/api/v1/strings`;

  try {
    const strings = yield call(request, requestURL);
    yield put(stringsLoaded(strings));
  } catch (err) {
    yield put(stringsLoadingError(err));
  }
}

export default function* stringData() {
  yield takeLatest(LOAD_STRINGS, getStrings);
}
