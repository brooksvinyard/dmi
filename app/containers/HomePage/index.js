/*
 * HomePage
 *
 * This is the Home Page that displays all the strings, at the '/' route
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import useInjectReducer from 'utils/injectReducer';
import useInjectSaga from 'utils/injectSaga';
import {
  makeSelectLoading,
  makeSelectError,
  makeSelectStrings,
} from 'containers/App/selectors';

import H2 from 'components/H2';
import P from 'components/P';
import Paper from 'components/Paper';
import StringsList from 'components/StringsList';

import saga from './saga';
import reducer from './reducer';
import { loadStrings } from '../App/actions';

export class HomePage extends React.Component {
  componentDidMount() {
    this.props.loadStrings();
  }

  render() {
    return (
      <Paper>
        <H2>Welcome to String Land!</H2>
        <P>Below are all the strings on the server.</P>

        <StringsList
          loading={this.props.loading}
          error={this.props.error}
          strings={this.props.strings}
        />
      </Paper>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  loading: makeSelectLoading(),
  error: makeSelectError(),
  strings: makeSelectStrings(),
});

HomePage.propTypes = {
  error: PropTypes.any,
  loading: PropTypes.bool,
  loadStrings: PropTypes.func,
  strings: PropTypes.any,
};

export function mapDispatchToProps(dispatch) {
  return {
    loadStrings: () => dispatch(loadStrings()),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = useInjectReducer({ key: 'homePage', reducer });
const withSaga = useInjectSaga({ key: 'homePage', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
  memo,
)(HomePage);
