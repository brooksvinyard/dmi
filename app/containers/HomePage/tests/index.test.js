import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';

// import { IntlProvider } from 'react-intl';

import HomePage from '../index';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
