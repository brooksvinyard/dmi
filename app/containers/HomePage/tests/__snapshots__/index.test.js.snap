// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  class="Paper-bhubno-0 bHtalL"
>
  <h2
    class="H2-egphza-0 ljvrU"
  >
    Welcome to String Land!
  </h2>
  <p
    class="P-sc-76wrpw-0 bTCnCC"
  >
    Below are all the strings on the server.
  </p>
  <div
    class="Error-qcpjhc-0 iRWWWH"
  >
    Something went wrong, please try again!
  </div>
</div>
`;
